### PR TITLE
Gov api return more docs

### PR DESCRIPTION
**Pull request by saisakul**
## Context

At the moment GOV search tool only return the top relevant page. However, it would be better to pull more relevant pages to use to produce answer - so that we can point to more references. 

## Changes proposed in this pull request

Query first 100 relevant pages from GOV search API. Then returns top 5 relevant pages based on cosine similarity.

## Guidance to review

It would be better if instead of hard coding the no. of pages we want the tool to return, to:
- have that as tool parameters.
- use similarity threshold to select no of documents. 

## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [ ] I have tested any code added or changed
- [ ] I have run integration tests

### CODE REVIEW

**Review by jamesrichards4 — COMMENTED** [review]
Let's pull both the retrieved_docs count and the returned_docs count out as parameters to the tool and set them based on a couple of new AISettings. That way we can experiment easily and they can just update their default settings to change this if they want

**Review by gecBurton — DISMISSED** [review]
v good idea

**Comment by saisakul** [comment]
> Let's pull both the retrieved_docs count and the returned_docs count out as parameters to the tool and set them based on a couple of new AISettings. That way we can experiment easily and they can just update their default settings to change this if they want

so rather pass them as additional parameters to the tools, I used the new AISettings from state that got passed during tool call.

File: redbox-core/redbox/models/chain.py [hunk 1]
```diff
@@ -78,6 +86,10 @@ class AISettings(BaseModel):
     # this is also the azure_openai_model
     chat_backend: ChatLLMBackend = ChatLLMBackend()
 
+    # settings for tool call
+    num_retrieve_results: int = 100
```
Comment (jamesrichards4) [inline redbox-core/redbox/models/chain.py:90]: could we give these more descriptive names? 

tool_govuk_retrieved_results
tool_govuk_returned_results

Comment (saisakul) [reply redbox-core/redbox/models/chain.py]: updated

**Review by jamesrichards4 — APPROVED** [review]

**Pull request merged**